src/vcpkg/commands.build.cpp

```diff
@@ -1168,6 +1168,22 @@ namespace vcpkg
         constexpr int max_port_file_count = 100;
 
         std::string portfile_cmake_contents;
+        std::vector<Path> files;
+        std::vector<std::string> hashes;
+
+        size_t i = 0;
+        for (auto& filestr : abi_info.pre_build_info->hash_additional_files)
+        {
+            Path file(filestr);
+            if (file.is_relative() || !fs.is_regular_file(file))
+            {
+                Checks::msg_exit_with_message(VCPKG_LINE_INFO, msgInvalidValueHashAdditionalFiles, msg::path = file);
+            }
+            const auto hash =
+                vcpkg::Hash::get_file_hash(fs, file, Hash::Algorithm::Sha256).value_or_exit(VCPKG_LINE_INFO);
+            abi_tag_entries.emplace_back(fmt::format("additional_file_{}", i++), hash);
```

Comment:
```suggestion
            auto hash =
                vcpkg::Hash::get_file_hash(fs, file, Hash::Algorithm::Sha256).value_or_exit(VCPKG_LINE_INFO);
            abi_tag_entries.emplace_back(fmt::format("additional_file_{}", i++), std::move(hash));
```

Reply:
From my experience that does not make a difference in assembly:
https://godbolt.org/z/93cPsq4vE

The compiler just optimizes both to the same. I personally find the `const` version better since it better reflects the fact that `hash` is just a constant helper variable which does not get mutated in this code section and you don't have to think about it. With `std::move` you have to think about the following: Does `hash` get mutated for some reason? Does `hash` get used after `std::move()` ?

Reply:
> Does `hash` get used after `std::move()` ?

Not really because it's the last line in that scope.

You could also get rid of the variable entirely.

Reply:
My point is that you need to think about it. If it is just `const auto` or `const auto&` you can directly see that it is a helper variable. Simply said. `const auto` has, in my opinion, less complexity then `std::move()` from a readers point of view. 

> You could also get rid of the variable entirely.

Yes and No. The compiler doesn't care but we are humans after all:
`abi_tag_entries.emplace_back(fmt::format("additional_file_{}", i++), hash);` already has two layers of complexity and 
`vcpkg::Hash::get_file_hash(fs, file, Hash::Algorithm::Sha256).value_or_exit(VCPKG_LINE_INFO);` also has two. Merging them into a single line/call increases (reading/understanding) complexity by a lot. It is like writing a sentence which ist simply too long with too many logical side parts. 

However, since the compiler doesn't care, I'll do whatever the tool maintainers will suggest.